File: 234-palindrome-linked-list/234-palindrome-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function isPalindrome(head: ListNode | null): boolean {
  let slow = head,
      fast = head,
      stack = [];
  
  while(fast && fast.next) {
    stack.push(slow.val);
    slow = slow.next;
    fast = fast.next;
    fast = fast.next;
  }
  
  if(fast)
    slow = slow.next;

  while(slow) {
    if(stack[stack.length - 1] != slow.val)
      return false;
    stack.pop();
    slow = slow.next;
  }
  
  return true;
};

File: 234-palindrome-linked-list/234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
	public boolean isPalindrome(ListNode head) {
		ListNode slow = head, fast = head, prev = null;

		while (fast != null) {
			ListNode temp = slow;
			slow = slow.next;

			if (fast.next != null)
				fast = fast.next.next;
			else
				break;
			
			temp.next = prev;
			prev = temp;
		}

		while (slow != null) {
			if (slow.val != prev.val)
				return false;
			slow = slow.next;
			prev = prev.next;
		}

		return true;
	}
}